src/main/java/com/powsybl/openloadflow/sa/AbstractSecurityAnalysis.java

```diff
@@ -133,32 +146,102 @@ SecurityAnalysisReport runSync(SecurityAnalysisParameters securityAnalysisParame
         findAllBranchesToClose(network, actions, topoConfig);
 
-        // load contingencies
-        List<Contingency> contingencies = contingenciesProvider.getContingencies(network);
         // try to find all switches impacted by at least one contingency and for each contingency the branches impacted
         PropagatedContingencyCreationParameters creationParameters = new PropagatedContingencyCreationParameters()
                 .setContingencyPropagation(securityAnalysisParametersExt.isContingencyPropagation())
                 .setShuntCompensatorVoltageControlOn(isShuntCompensatorVoltageControlOn(lfParameters))
                 .setSlackDistributionOnConformLoad(lfParameters.getBalanceType() == LoadFlowParameters.BalanceType.PROPORTIONAL_TO_CONFORM_LOAD)
                 .setHvdcAcEmulation(lfParameters.isHvdcAcEmulation());
 
-        List<PropagatedContingency> propagatedContingencies = PropagatedContingency.createList(network, contingencies, topoConfig, creationParameters);
+        SecurityAnalysisResult result;
+        if (securityAnalysisParametersExt.getThreadCount() == 1) {
+            List<PropagatedContingency> propagatedContingencies = PropagatedContingency.createList(network, contingencies, topoConfig, creationParameters);
 
-        var parameters = createParameters(lfParameters, lfParametersExt, topoConfig.isBreaker());
+            var parameters = createParameters(lfParameters, lfParametersExt, topoConfig.isBreaker());
 
-        // create networks including all necessary switches
-        try (LfNetworkList lfNetworks = Networks.load(network, parameters.getNetworkParameters(), topoConfig, saReportNode)) {
-            // run simulation on largest network
-            SecurityAnalysisResult result = lfNetworks.getLargest().filter(n -> n.getValidity() == LfNetwork.Validity.VALID)
-                    .map(largestNetwork -> runSimulations(largestNetwork, propagatedContingencies, parameters, securityAnalysisParameters, operatorStrategies, actions, limitReductions))
-                    .orElse(createNoResult());
+            // create networks including all necessary switches
+            try (LfNetworkList lfNetworks = Networks.load(network, parameters.getNetworkParameters(), topoConfig, saReportNode)) {
+                // run simulation on largest network
+                result = lfNetworks.getLargest().filter(n -> n.getValidity() == LfNetwork.Validity.VALID)
+                        .map(largestNetwork -> runSimulations(largestNetwork, propagatedContingencies, parameters, securityAnalysisParameters, operatorStrategies, actions, limitReductions))
+                        .orElse(createNoResult());
+            }
+        } else {
+            if (!network.getVariantManager().isVariantMultiThreadAccessAllowed()) {
+                throw new PowsyblException("To run a security analysis on multiple threads (and so on variants), you need to set allowVariantMultiThreadAccess to true");
+            }
```

Comment:
Java API users will make it, but wondering how to make this more user friendly, in particular for PyPowSyBl users.

Any way to make this transparent / working out of the box ?
Or should we add an `iidm` platform module config so that variant manager initializes with multithread access enabled ?

Reply:
We could automatically set this mode when running the multi threaded SA but that might be dangerous and not consistent with what we have done until today to change a network model when running a security. I think users expect that the SA only read data and not change anything in the data model.

So yes maybe we could just allow to configure the IIDM multithread access default value using the platform config.

Reply:
A security analysis mode, we do not update the network (IIDM) with N state.